api/apps/api/src/modules/scenarios/locks/lock.service.spec.ts

```diff
@@ -0,0 +1,226 @@
+import { Test } from '@nestjs/testing';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { EntityManager } from 'typeorm';
+import * as faker from 'faker';
+
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import {
+  lockedScenario,
+  lockNotFoundError,
+  LockService,
+} from '@marxan-api/modules/scenarios/locks/lock.service';
+import {
+  JobStatus,
+  Scenario,
+  ScenarioType,
+} from '@marxan-api/modules/scenarios/scenario.api.entity';
+import { ScenarioLockEntity } from '@marxan-api/modules/scenarios/locks/scenario.lock.entity';
+import { User } from '@marxan-api/modules/users/user.api.entity';
+import { left, right } from 'fp-ts/Either';
+
+let fixtures: FixtureType<typeof getFixtures>;
+
+describe('Lock service test suite', () => {
+  beforeEach(async () => {
+    fixtures = await getFixtures();
+  });
+
+  it(`should be able to grab lock if lock is available`, async () => {
+    await fixtures.GivenNoLockHasBeenGrabbed();
+    await fixtures.ThenShouldBeAbleToGrabNewLockForGivenScenario();
+  });
+
+  it(`should not be able to grab lock if lock is not available`, async () => {
+    await fixtures.GivenALockHasBeenGrabbed();
+    await fixtures.ThenShouldNotBePossibleToGrabNewLockForGivenScenario();
+  });
+
+  it(`should be able to release the lock if lock exists`, async () => {
+    await fixtures.GivenALockHasBeenGrabbed();
+    await fixtures.ThenSameUserShouldBeAbleToReleaseLock();
+  });
+
+  it(`should not be able to release the lock if lock exists for a different user`, async () => {
+    await fixtures.GivenNoLockHasBeenGrabbed();
+    await fixtures.ThenOtherUserShouldNotBeAbleToReleaseLock();
+  });
+
+  it(`should not be able to release the lock if lock does not exist`, async () => {
+    await fixtures.GivenNoLockHasBeenGrabbed();
+    await fixtures.ThenShouldNotBeAbleToReleaseLock();
+  });
+
+  it(`isLocked should return true if a lock exists`, async () => {
+    await fixtures.GivenALockExists();
+    await fixtures.ThenIsLockedReturnsTrue();
+  });
+
+  it(`isLocked should return false if lock no lock exists`, async () => {
+    await fixtures.GivenNoLockExists();
+    await fixtures.ThenIsLockedReturnsFalse();
+  });
+});
+
+async function getFixtures() {
+  const fakeUser: User = {
+    id: 'fake-user',
+    email: faker.internet.email(),
+    passwordHash: faker.random.uuid(),
+    isActive: true,
+    isDeleted: false,
+    projects: [],
+    scenarios: [],
+  };
```

Comment:
do we need whole User and reference it at all?

Reply:
nope, neither the scenario down below :ok_hand: